linera-views/src/rocks_db.rs

```diff
@@ -490,6 +607,20 @@ impl AdminKeyValueStore for RocksDbStore {
         Ok(Self { store })
     }
 
+    fn clone_with_root_key(&self, root_key: &[u8]) -> Result<Self, RocksDbStoreError> {
+        let store = self.inner_clone_with_root_key(root_key)?;
+        let cache_size = self.cache_size();
+        #[cfg(with_metrics)]
+        let store = MeteredStore::new(&ROCKS_DB_METRICS, store);
+        let store = ValueSplittingStore::new(store);
+        #[cfg(with_metrics)]
+        let store = MeteredStore::new(&VALUE_SPLITTING_METRICS, store);
+        let store = LruCachingStore::new(store, cache_size);
+        #[cfg(with_metrics)]
+        let store = MeteredStore::new(&LRU_CACHING_METRICS, store);
+        Ok(Self { store })
```

Comment:
And with an `fn from_inner` you could deduplicate this and `connect`:
```suggestion
        let store = self.inner_clone_with_root_key(root_key)?;
        Ok(Self::from_inner(store))
```

Reply:
Ok, this indeed led to some simplification.